generic/tests/netperf.py

```diff
@@ -74,13 +74,37 @@ def mtu_set(mtu):
         if netdst in br_in_use:
             ifaces_in_use = host_bridges.list_iface()
             target_ifaces = list(ifaces_in_use + br_in_use)
-        if params.get("netdst_nic1") in process.system_output(
-                "ovs-vsctl list-br", ignore_status=True, shell=True).decode():
-            ovs_list = "ovs-vsctl list-ports %s" % params["netdst_nic1"]
-            ovs_port = process.system_output(ovs_list,
-                                             shell=True).decode().splitlines()
-            target_ifaces = target_ifaces + \
-                params.objects("netdst_nic1") + ovs_port
+        if process.system("which ovs-vsctl && systemctl status openvswitch.service",
+                          ignore_status=True, shell=True) == 0:
+            ovs_br_all = netperf_base.ssh_cmd(host, "ovs-vsctl list-br")
+            ovs_br = []
+            if ovs_br_all:
+                for nic in vm.virtnet:
+                    if nic.netdst in ovs_br:
+                        ovs_br.append(nic.netdst)
+                    elif nic.nettype == "vdpa":
+                        vf_pci = netperf_base.ssh_cmd(
+                                host,
+                                "vdpa dev show |grep %s | grep -o 'pci/[^[:space:]]*' | awk -F/ '{print $2}'"
+                                % nic.netdst)
+                        pf_pci = netperf_base.ssh_cmd(
+                                host,
+                                "grep PCI_SLOT_NAME /sys/bus/pci/devices/%s/physfn/uevent | cut -d'=' -f2"
+                                % vf_pci)
+                        port = netperf_base.ssh_cmd(
+                                host,
+                                "ls /sys/bus/pci/devices/%s/net/ | head -n 1"
+                                % pf_pci)
+                        ovs_br_vdpa = netperf_base.ssh_cmd(host, "ovs-vsctl port-to-br %s" % port)
+                        cmd = "ovs-ofctl add-flow {} 'in_port=1,idle_timeout=0 actions=output:2'".format(ovs_br_vdpa)
+                        cmd += "&&  ovs-ofctl add-flow {} 'in_port=2,idle_timeout=0 actions=output:1'".format(ovs_br_vdpa)
+                        cmd += "&&  ovs-ofctl dump-flows {}".format(ovs_br_vdpa)
+                        netperf_base.ssh_cmd(host, cmd)
+                        ovs_br.append(ovs_br_vdpa)
+                for br in ovs_br:
+                    ovs_list = "ovs-vsctl list-ports %s" % br
+                    ovs_port = netperf_base.ssh_cmd(host, ovs_list)
+                    target_ifaces.extend([br + ovs_port])
```

Comment:
Both `br` and `ovs_port` are strings, if their names are `br0` and `eth0`, here `br + ovs+port` is `br0eht0`.
Suggest `target_ifaces.extend(ovs_list.split() + [br])`

Reply:
It should be target_ifaces.extend(ovs_port.split() + [br]). I have updated all the suggestion, please help  review.